enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,319 @@
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Enable the configuration of automated cluster backups.
+
+## Motivation
+
+The current documented procedure for performing a backup of an OpenShift
+cluster is manually initiated. This reduces the likelyhood of a timely backup
+being available in the event of catastrophic cluster failure.
+
+The procedure also requires gaining a root shell on a control plane
+node. Shell access to OpenShift control plane nodes access is generally
+discouraged due to the potential for affecting the reliability of the
+node. Indeed, at least one customer reports having had operator error result
+in a control plane node being taken offline in the process of attempting to
+follow the documented backup procedure.
+
+Finally, OpenShift 4.9 is intended to ship with etcd 3.5, and this will mean
+an upgrade for existing clusters from 3.4. Ensuring support for automated
+backups in the 4.8 edge required to upgrade to 4.9 will increase the
+likelyhood that customers will have cluster backups available in case of any
+trouble with the etcd upgrade.
+
+### Goals
+
+1. One-time cluster backup can be initiated without requiring a root shell.
+1. Automated backups can be configured as a day 2 operation.
+1. Cluster backups are written to the host filesystem of control plane nodes.
+
+### Non-Goals
+
+1. Backup to cluster (i.e. PVC) or cloud storage (e.g. S3).
+1. Support day 1 configuration of automated backups.
+1. Automate cluster restoration.
+1. Rewrite the existing cluster backup script.
+
+## Proposal
+
+- Add a new API singleton supporting backup configuration
+  - Suggested resource name `backups.config.openshift.io`
+    - Separate backup configuration from the etcd operator in case
+      implementation migrates to a dedicated backup operator.
+  - Fields:
+    - spec.reason `string`
+      - Set to a value different from the most recent reason, will prompt a
+        one-time backup.
+      - Default: `""`
+    - spec.schedule `string`
+      - [Standard cron expression](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax)
+      - Results in backups scheduled repeatedly to successive control plane
+        nodes.
+      - Default: `""`
+    - spec.maxDurationWithoutBackup `duration`
+      - If a backup has not successfully completed in the configured
+        interval, a critical alert will be generated for this and every
+        subsequent interval. The alert will either indicate that backups are
+        not configured, or supply the most recent reason a backup failed.
+      - Setting to `0` will disable alerting.
+      - Default: `1h`
```

Comment:
1h? why do u need such high frequency?

Reply:
The major drawback is in case something is corrupted, or a user caused some issue and will notice only the next day , you'll continuously  override all old copies.
Doesn't 1d make more sense?

Reply:
- We're not intending to enable backups by default. The user will be prompted (by the 'backup not completed within x amount of time alert) to configure backups to a configuration appropriate to their needs when the cluster upgrades to a version that supports it.
- The number we set as a default is pretty arbitrary at this point - it's a placeholder. I would expect to solicit input from actual users rather than any of us assuming we know best. 
- A backup that is more than a short interval old would be problematic if the cluster used cluster storage. Most csi plugins use the api as their source of truth. In the event of _any_ storage changes occurring since the backup was taken, it is likely that the api state wouldn't match the storage state and reconciliation could be difficult or even impossible.

Reply:
Depending on what you mean, it seems that this default may have to auto-adjust

Reply:
@deads2k Would you please elaborate as to why a performance-impacting action should change without human input?